### PR TITLE
Fix NJT stride access in SDPA dispatcher logic

### PR DESCRIPTION
**Pull request by jbschlosser**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #117860
* __->__ #119846
* #119910

`._stride` -> `._strides`

Adds test to cover this case.

### CODE REVIEW

File: test/test_nestedtensor.py [hunk 1]
```diff
@@ -3903,6 +3906,51 @@ def test_sdpa_with_constant_sequence_length(self, device, dtype):
         output_dense = F.scaled_dot_product_attention(query._values, key._values, value._values)
         self.assertEqual(output._values, output_dense)
 
+    @onlyCUDA
+    @unittest.skipIf(
+        not PLATFORM_SUPPORTS_FLASH_ATTENTION and not PLATFORM_SUPPORTS_MEM_EFF_ATTENTION,
```
Comment (drisspg) [inline test/test_nestedtensor.py:3911]: nit there is another one that contains both, PLATFORM_SUPPORTS_FUSED_ATTENTION

File: torch/nested/_internal/sdpa.py [hunk 2]
```diff
@@ -290,6 +290,7 @@ def _select_sdp_backend(query, key, value, attn_mask, dropout, is_causal):
             if math_sdp_enabled() and _can_use_math_sdpa_jagged(params):
                 return SDPBackend.MATH
 
+    breakpoint()
```
Comment (drisspg) [inline torch/nested/_internal/sdpa.py:293]: 😜

**Review by drisspg — APPROVED** [review]
I trust you

File: torch/nested/_internal/sdpa.py (hunk 2 above)
Comment (jbschlosser) [reply torch/nested/_internal/sdpa.py:293]: this is what happens when you rush :(

**Review by ani300 — APPROVED** [review]
lgtm too!

**Review by soulitzer — APPROVED** [review]
nice catch!

**Pull request closed without merging**